File: android/src/main/kotlin/com/maktubcompany/easy_blue_printer/plugin/domain/usecase/PrintImageUseCase.kt
package com.maktubcompany.easy_blue_printer.plugin.domain.usecase

import com.maktubcompany.easy_blue_printer.plugin.domain.entities.BluetoothDeviceEntity
import com.maktubcompany.easy_blue_printer.plugin.domain.repository.BluetoothRepository

class PrintImageUseCase(private val repository: BluetoothRepository) {
    fun execute(data: ByteArray, align: Int): Boolean {
        return repository.printImage(data, align)
    }
}

File: ios/easy_blue_printer/Sources/easy_blue_printer/plugin/domain/usecases/PrintImageUseCase.swift
import Foundation

public class PrintImageUseCase {
    private let repository: BluetoothRepository

    public init(repository: BluetoothRepository) {
        self.repository = repository
    }

    public func execute(data: Data, align: Int) -> Bool {
        return repository.printImage(data: data, align: align)
    }
}
